src/otx/core/model/module/anomaly/anomaly_lightning.py

```diff
@@ -0,0 +1,450 @@
+        pixel_threshold: float = 0.5,
+        task: TaskType = TaskType.CLASSIFICATION,
+        # the actual values for mean and scale should be in range 0-255
+        mean_values: tuple[float, float, float] = (0.0, 0.0, 0.0),
+        scale_values: tuple[float, float, float] = (1.0, 1.0, 1.0),
+        normalization_scale: float = 1.0,
+    ) -> None:
+        self.orig_height, self.orig_width = image_shape
+        metadata = {
+            ("model_info", "image_threshold"): image_threshold,
+            ("model_info", "pixel_threshold"): pixel_threshold,
+            ("model_info", "normalization_scale"): normalization_scale,
+            ("model_info", "orig_height"): image_shape[0],
+            ("model_info", "orig_width"): image_shape[1],
+            ("model_info", "image_shape"): image_shape,
+            ("model_info", "labels"): "Normal Anomaly",
+            ("model_info", "model_type"): "AnomalyDetection",
+            ("model_info", "task"): task.value,
+        }
+        super().__init__(
+            input_size=(1, 3, *image_shape),
+            mean=mean_values,
+            std=scale_values,
+            swap_rgb=False,  # default value. Ideally, modelAPI should pass RGB inputs after the pre-processing step
+            metadata=metadata,
+        )
+
+    def to_openvino(
+        self,
+        model: nn.Module,
+        output_dir: Path,
+        base_model_name: str = "exported_model",
+        precision: OTXPrecisionType = OTXPrecisionType.FP32,
+    ) -> Path:
+        save_path = str(output_dir / f"{base_model_name}.xml")
+        exported_model = openvino.convert_model(
+            input_model=model,
+            example_input=torch.rand(self.input_size),
+            input=(openvino.runtime.PartialShape(self.input_size)),
+        )
+        exported_model = self._postprocess_openvino_model(exported_model)
+        openvino.save_model(exported_model, save_path, compress_to_fp16=(precision == OTXPrecisionType.FP16))
+        return Path(save_path)
+
+    def to_onnx(
+        self,
+        model: nn.Module,
+        output_dir: Path,
+        base_model_name: str = "exported_model",
+        precision: OTXPrecisionType = OTXPrecisionType.FP32,
+        embed_metadata: bool = True,
+    ) -> Path:
+        save_path = str(output_dir / f"{base_model_name}.onnx")
+        torch.onnx.export(
+            model=model,
+            args=(torch.rand(1, 3, self.orig_height, self.orig_width)).to(
+                next(model.parameters()).device,
+            ),
+            f=save_path,
+            opset_version=14,
+            dynamic_axes={"input": {0: "batch_size"}, "output": {0: "batch_size"}},
+            input_names=["input"],
+            output_names=["output"],
+        )
+        onnx_model = onnx.load(save_path)
+        onnx_model = self._postprocess_onnx_model(onnx_model, embed_metadata, precision)
+        onnx.save(onnx_model, save_path)
+        return Path(save_path)
+
+
+class OTXAnomaly:
```

Comment:
Can we name this as mixin? I was initially confused that this was the actual implementation